### PR TITLE
Refactors beds stools chairs, and adds a few cool things

### LINKED ISSUE #9703: [Delta] Random double airlock
Random double airlock here. Just south of arrivals.

dreamseeker_2018-10-04_18-33-27

### PR DESCRIPTION
**Pull request by SamHPurp**
INFO FOR CODERS: Full refactor of all chairs beds and stools. /stool/bed/chair is now /chair... /stool/bed is now /bed... /stool is now /stool

I've moved the chair sprites into their own dmi for ease of use.

Added a post-unbuckle proc to mirror the post-buckle proc

Also added in dog beds for Runtime & Renault, but didn't map them in.

INFO FOR PLAYERS: You can now pick up some chairs and stools, and beat people with them. They have a chance to break while you're beating people with it.

Adds new bar stools, specifically for the bar. Sprite credit to TG.

image 
image

Also fixes #9703

:cl: Purpose
add: You can now pick up most chairs and beat colleagues with them. Careful too many whacks and they'll break.
add: Adds new bar stools.
fix: Delta - Removes extra door in arrivals.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/stool_bed_chair_nest/stools.dm [hunk 1]
```diff
@@ -1,77 +0,0 @@
...
-		qdel(src)
-
-/obj/structure/stool/attackby(obj/item/W as obj, mob/user as mob, params)
-	if(istype(W, /obj/item/wrench))
-		playsound(loc, W.usesound, 50, 1)
-		new buildstacktype(loc, buildstackamount)
-		qdel(src)
-
-/obj/structure/stool/MouseDrop(atom/over_object, src_location, over_location, src_control, over_control, params, skip_fucking_stool_shit = 0)
-	if(skip_fucking_stool_shit)
```
Comment (SamHPurp) [inline code/game/objects/structures/stool_bed_chair_nest/stools.dm]: rip best variable

**Comment by Ziiro** [comment]
absolute fucking madman
>lets not pick up electric chairs either

throwing an electric chair should not only be possible, but encouraged. Chance to stun on hit like when you throw a stunbaton. 😂

**Comment by KasparoVy** [comment]
Stunbatonthrowing works here?

**Comment by tigercat2000** [comment]
This is obviously better than what we have now, but I'm curious about making beds, stools, and chairs all individual /structure's, rather than having any sort of more related common parent. Is that little behavior really shared?

**Comment by SamHPurp** [comment]
@tigercat2000 I was thinking about this as well... in the end I figured there is enough common functionality (picking things up etc) that it was worth parenting stools to chair.

I can change it if required. Will just take a long time for the map edits.

**Comment by SamHPurp** [comment]
> Stunbatonthrowing works here?

afaik no.

**Comment by KasparoVy** [comment]
Thanks, I figured as much but it's been so long since I last tried I was unsure.

**Comment by taukausanake** [comment]
The new stools are nice but I like the old ones too. Also, pretty sure we have that stool spacepod somewhere. Kind of sad to see it not match the new stools

**Comment by SamHPurp** [comment]
>The new stools are nice but I like the old ones too.

The old stools are still there. These are simply just for the ones in the Cyberiad bar.

**Comment by SamHPurp** [comment]
@variableundefined let me know when you're looking to merge this.... and I'll resolve conflicts then.... rather than do several waves of conflict resolution...

**Comment by KasparoVy** [comment]
Smart strat

**Comment by variableundefined** [comment]
This PR's approved, will need to wait till I (Or another maintainer) is available to review and merge before asking you to fix conflicts though.

**Pull request merged**